pkg/controller/cyclenoderequest/transitioner/transitions.go

```diff
@@ -517,28 +518,37 @@ func (t *CycleNodeRequestTransitioner) transitionHealing() (reconcile.Result, er
 		return t.transitionToFailed(err)
 	}
 
-	// try and re-attach the nodes, if any were un-attached
 	for _, node := range t.cycleNodeRequest.Status.NodesToTerminate {
+		// nodes in NodesToTerminate may have been terminated, so check if they still exist
+		if nodeExists, err := k8s.NodeExists(node.Name, t.rm.RawClient); err != nil || !nodeExists {
+			t.rm.LogEvent(t.cycleNodeRequest,
+				"HealingNodes", "Skip healing node: %s, err: %v",
+				node.Name, err)
+			continue
+		}
+
+		// try and re-attach the nodes, if any were un-attached
 		t.rm.LogEvent(t.cycleNodeRequest, "AttachingNodes", "Attaching instances to nodes group: %v", node.Name)
 		alreadyAttached, err := nodeGroups.AttachInstance(node.ProviderID, node.NodeGroupName)
 		if alreadyAttached {
 			t.rm.LogEvent(t.cycleNodeRequest,
 				"AttachingNodes", "Skip re-attaching instances to nodes group: %v, err: %v",
 				node.Name, err)
-			continue
 		}
 		if err != nil {
-			return t.transitionToFailed(err)
+			t.rm.LogWarningEvent(t.cycleNodeRequest,
```

Comment:
Why change this to a log, don't we want cyclops to transitioned to failed immediately? Otherwise we're in a weird state where we don't know if it should proceed or not.

Reply:
As I understand, if it `transitionToFailed` here, it will skip all the rest nodes in the loop, which probably is not expected. For example, if the first node in `NodesToTerminate` has already been successfully cycled and terminated, it will get an error here and return, no more nodes will be looped here to re-attach and uncordon.

Reply:
Right then you should intercept the error and ensure that that is the case. Logging on every possible error will eventually have unintended consequences. We already do something similar here: https://github.com/atlassian-labs/cyclops/blob/master/pkg/cloudprovider/aws/aws.go#L21

Try to implement some version of that logic in here to ensure it only logs out when you expect it to, otherwise transition to failed because that's not an expected outcome of the request.

Reply:
Cool, I've updated it so that it will return on errors.